pkgs/development/compilers/rust/make-rust-platform.nix

```diff
@@ -75,6 +86,19 @@
         bindgenHook
         ;
     };
+in
+(makeScopeWithSplicing' {
+  otherSplices = {
+    selfBuildBuild = lib.makeScope pkgsBuildBuild.newScope f;
+    selfBuildHost = lib.makeScope pkgsBuildHost.newScope f;
+    selfBuildTarget = lib.makeScope pkgsBuildTarget.newScope f;
+    selfHostHost = lib.makeScope pkgsHostHost.newScope f;
+    selfHostTarget = lib.makeScope pkgsHostTarget.newScope f;
+    selfTargetTarget = lib.optionalAttrs (pkgsTargetTarget ? newScope) (
+      lib.makeScope pkgsTargetTarget.newScope f
+    );
+  };
+  inherit f;
 })
```

Comment:
This avoid coupling with `pkgs.rustPlatform` introduced with `generateSplicesForMkScope "rustPlatform"`.